ksqldb-tools/src/test/java/io/confluent/ksql/tools/migrations/MigrationsTest.java

```diff
@@ -88,8 +96,44 @@ public static void classTearDown() {
   }
 
   @Test
-  public void dummy() {
-    // placeholder until additional functionality (beyond initialization) is added
+  public void testApply() throws IOException {
+    // Migration file
+    createMigrationFile(
+        1,
+        "foo",
+        MigrationsDirectoryUtil.getMigrationsDirFromConfigFile(configFilePath),
+        "CREATE STREAM FOO (A STRING) WITH (KAFKA_TOPIC='FOO', PARTITIONS=1, VALUE_FORMAT='DELIMITED');"
+    );
+    createMigrationFile(
+        2,
+        "bar",
+        MigrationsDirectoryUtil.getMigrationsDirFromConfigFile(configFilePath),
+        "CREATE STREAM BAR (A STRING) WITH (KAFKA_TOPIC='BAR', PARTITIONS=1, VALUE_FORMAT='DELIMITED');"
+    );
+    final int status = MIGRATIONS_CLI.parse("--config-file", configFilePath, "apply").run();
+    assertThat(status, is(0));
+
+    // verify FOO and BAR were registered
+    describeSource("FOO");
+    describeSource("BAR");
+
+    // verify current
+    final List<String> current = makeKsqlQuery("SELECT * FROM migration_schema_versions WHERE VERSION_KEY='CURRENT';");
+    assertThat(current.size(), is(2));
```

Comment:
Why do we get two result rows from this query? It's a pull query, right? Shouldn't that only return the latest row? I feel like there's something I don't understand about pull queries.

Reply:
This queries the websocket endpoint, which returns the column names as the first item in the list.